functions/go/set-image/custom/custom.go

```diff
@@ -0,0 +1,43 @@
+package custom
+
+import (
+	"fmt"
+
+	"github.com/GoogleContainerTools/kpt-functions-sdk/go/fn"
+	"sigs.k8s.io/kustomize/api/filters/imagetag"
+	"sigs.k8s.io/kustomize/api/types"
+	"sigs.k8s.io/kustomize/kyaml/filtersutil"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+func SetAdditionalFieldSpec(image types.Image, objects fn.KubeObjects, additionalImageFields types.FsSlice, ctx *fn.Context) {
```

Comment:
nit: can you add a docstring to explain what `SetAdditionalFieldSpec` does. and the long term plans we discussed why and when additionalFieldSpec should be dropped.